src/main/java/org/openstreetmap/atlas/geography/atlas/sqlite/SQLiteItemWriter.java

```diff
@@ -0,0 +1,216 @@
+import org.openstreetmap.atlas.utilities.maps.LongToLongMap;
+
+/**
+ * To save all of the data within an atlas, we need to get each {@link AtlasItem} type and iterate
+ * though them with inserts into the database. This class declares that a given writer needs to get
+ * the atlas to do its job.
+ *
+ * @author hallahan
+ * @author lcram
+ */
+public class SQLiteItemWriter
+{
+    private static final String INSERT_SQL = "INSERT INTO %s VALUES (?, ?, ?, ?)";
+    private static final String INSERT_TAGS_SQL = "INSERT INTO %s_tags VALUES (?, ?, ?)";
+    private static final String INSERT_RTREE_SQL = "INSERT INTO %s_rtree VALUES (?, ?, ?, ?, ?)";
+
+    // database column constants
+    private static final int FIRST_COLUMN = 1;
+    private static final int SECOND_COLUMN = 2;
+    private static final int THIRD_COLUMN = 3;
+    private static final int FOURTH_COLUMN = 4;
+    private static final int FIFTH_COLUMN = 5;
+
+    private final Connection connection;
+    private final PreparedStatement itemStatement;
+    private final PreparedStatement tagsStatement;
+    private final Optional<PreparedStatement> rtreeStatementOptional;
+
+    private final LongToLongMap identifierToTagStoreIndex;
+    private final PackedTagStore tagStore;
+    private final ItemType itemType;
+
+    SQLiteItemWriter(final Connection connection, final boolean buildIndices,
+            final PackedTagStore tagStore, final LongToLongMap identifierToTagStoreIndex,
+            final ItemType itemType) throws SQLException
+    {
+        this.tagStore = tagStore;
+        this.identifierToTagStoreIndex = identifierToTagStoreIndex;
+        this.itemType = itemType;
+        this.connection = connection;
+
+        // Construct the prepared statements
+        this.itemStatement = connection.prepareStatement(
+                String.format(INSERT_SQL, this.itemType.toString().toLowerCase()));
+        this.tagsStatement = connection.prepareStatement(
+                String.format(INSERT_TAGS_SQL, this.itemType.toString().toLowerCase()));
+        if (buildIndices)
+        {
+            this.rtreeStatementOptional = Optional.of(connection.prepareStatement(
+                    String.format(INSERT_RTREE_SQL, this.itemType.toString().toLowerCase())));
+        }
+        else
+        {
+            this.rtreeStatementOptional = Optional.empty();
+        }
+    }
+
+    public void write(final PackedAtlas atlas) throws SQLException
+    {
+        // This is very important. When you're doing bulk inserts, you need to turn this off.
+        // Performance is very bad without doing this!
+        this.connection.setAutoCommit(false);
+
+        final Iterable<? extends AtlasItem> items = getItemIterableOfCorrectType(atlas,
+                this.itemType);
+
+        for (final AtlasItem item : items)
+        {
+            // INSERT feature data
+            final long atlasId = item.getIdentifier();
```

Comment:
this may not be possible and/or efficient, but I think you could add a `getItemSQL()` to each item class, then you could just maintain it with the data itself instead of in a separate set of switches and generic logic. e.g., inside of Node.java, 
```Java
public PreparedStatement getItemSql(Connection connection, String shardName) {
    String itemStatement = connection.prepareStatement(
                 String.format(SQLiteItemWriter.INSERT_SQL, this.itemType.toString().toLowerCase()));
    itemStatement.setLong(SQLiteItemWriter.FIRST_COLUMN, this.getIdentifier());
    itemStatement.setString(SQLiteItemWriter.SECOND_COLUMN, "md5sum_here");
    itemStatement.setString(SQLiteItemWriter.THIRD_COLUMN, shardName);
    itemStatement.setBytes(SQLiteItemWriter.FOURTH_COLUMN,
        new WkbLocationConverter().convert(this.getLocation()));
    return itemStatement;
}
```
Same thing for tag statement, which could be in AtlasEntity.java. Then, you could just iterate over each atlas entity directly, skipping over unrequested / unsupported types, like 
```
for (final AtlasItem item : entitiesIterator) {
    if( checkEntityTypeSupported(item)) {
        item.getItemSQL(connection, shardName).execute(); 
    } 
}
```
Obvs this is scratch code and just a thought! May not be worth the refactor or even actually work.